code/game/machinery/bots/farmbot.dm

```diff
@@ -565,9 +565,6 @@
 		src.build_step++
 		to_chat(user, "You complete the Farmbot! Beep boop.")
 		var/obj/machinery/bot/farmbot/S = new /obj/machinery/bot/farmbot
-		for ( var/obj/structure/reagent_dispensers/watertank/wTank in src.contents )
```

Comment:
И всё таки, сколько ватертанков было внутри? И почему убираешь сброс тут, или там где-то еще дробконтент идёт?

Reply:
Короче, почему их было три. 
1. Изначально ватертанк добавляется в контентсы, при спавне фармбота. На случай, если он будет защитспавнен админами. 
V вот эту проверку он не проходил. Ибо был эссембли, а не сам фармбот. 
![image](https://user-images.githubusercontent.com/12589405/36378291-b54ffd24-1593-11e8-9ef8-d5912a2b4a4e.png)

2.  При добавлении роборуки он, собственно, не удалялся, а перетекал в контентсы. 
(src.loc = A //Place the water tank into the assembly, it will be needed for the finished bot) (на этом этапе их становилось в контентсах уже два)
3. Все ватертанки, что были в эссембли, переходили в фармбота. (их было два плюс один добавлялся при спавне самого фармбота - три). Я же в этом месте убрала их перенос во время создания фармбота. Теперь если на любом этапе разбить его, ватертанк будет 1 и никуда не пропадет. Во всем процессе создания фарбота он был единственным. 
Итого они брались: 
![image](https://user-images.githubusercontent.com/12589405/36378592-da76d0f4-1594-11e8-86af-ec4284549c48.png)
^ раз 
![image](https://user-images.githubusercontent.com/12589405/36378624-f76bfb12-1594-11e8-8b23-cf436846c74e.png)
^ два
![image](https://user-images.githubusercontent.com/12589405/36378709-43ac8398-1595-11e8-8115-d19ca994cefa.png)
^ три

Тесты показали, что если разбить щитспавненого фармбота, выпадал один ватертанк. При разбиении собранного - три.